src/memory/sparse.rs

```diff
@@ -193,6 +193,32 @@ impl<R: Register> Memory for SparseMemory<R> {
         Ok(())
     }
 
+    fn load_bytes(&mut self, addr: u64, length: usize) -> Result<Bytes, Error> {
+        if length == 0 {
+            return Ok(Bytes::new());
+        }
+        let mut current_page_addr = round_page_down(addr);
+        let mut current_page_offset = addr - current_page_addr;
+        let mut need_read_len = length;
+        let mut out_value = Vec::new();
+        out_value.resize(length, 0);
```

Comment:
I don't recommend pre-allocating memory for vec here:

In the design of syscall set_content, the value of length comes from the user, so we have to consider a question, if the user enters a wrong length, how should ckb-vm respond.

In your code, if length is a large number, ckb-vm will oom, with disastrous results.

Reply:
Therefore I think it is better to use Vec::extend to dynamically add content